forem/forem
# fast follow refactor, handle downthread blocked commenters

## What type of PR is this? (check all applicable)


## Description

This addresses the feedback from https://github.com/forem/forem/pull/19898, where we stopped commenters from replying on a comment where the article writer was not blocking but the author of the immediate parent comment was.

In addition to cleaning up some expectations in the spec and making the relevant controller method more readable, this change also prevents a commenter multiple comments downthread of a blocking user from leaving their comment.

## Related Tickets & Documents



- Related Issue #19878
- Closes #19878

## QA Instructions, Screenshots, Recordings
- log in as admin, grab five user emails by visiting Admin -> Member Manager -> Member show pages
- log in as user 1, search for user 2’s username, visit their profile and block them
- find an article in the feed written by user 3, and have user 1 leave a comment on it.
- log in as user 4, visit the commented-on article, leave a comment on user 1's comment
- log in as user 5, visit the article, and leave a comment on user 4's comment
- log in as (blocked) user 2 and try to leave a comment under user 5’s comment on the article
- an alert should notify the user that moderation has prevented the comment from POSTing

### UI accessibility checklist
_If your PR includes UI changes, please utilize this checklist:_

_For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?

      have not been included_

## [optional] What gif best describes this PR or how it makes you feel?

![guy holding out "halt" hand and gesturing at ID armband to indicate the viewer is unauthorized to proceed](https://media.tenor.com/4ijO8E1idGUAAAAC/bash-the-entertainer-door-security.gif)
